src/imitation/scripts/train_imitation.py

```diff
@@ -140,6 +141,41 @@ def dagger(
     return stats
 
 
+@train_imitation_ex.command
+def sqil(
+    sqil: Mapping[str, Any],
+    _run,
+    _rnd: np.random.Generator,
+) -> Mapping[str, Mapping[str, float]]:
+    custom_logger, log_dir = logging_ingredient.setup_logging()
+    expert_trajs = demonstrations.get_expert_trajectories()
+
+    with environment.make_venv() as venv:
+        sqil_trainer = SQIL(
+            venv=venv,
+            demonstrations=expert_trajs,
+            policy=sqil["policy_model"],
+            custom_logger=custom_logger,
+            rl_algo_class=sqil["rl_algo_class"],
+            rl_kwargs=sqil["rl_kwargs"],
+        )
+
+        sqil_trainer.train(
+            total_timesteps=int(sqil["total_timesteps"]),
+            **sqil["train_kwargs"],
+        )
+        sqil_trainer.save_policy(policy_path=osp.join(log_dir, "final.th"))
+
+        imit_stats = policy_evaluation.eval_policy(sqil_trainer.policy, venv)
+
+    stats = {"imit_stats": imit_stats}
```

Comment:
A lot of boilerplate code here (almost identical to the end of dagger and bc functions) -- can we reduce code duplication some way?

Reply:
Made a shared function called `_collect_stats`!

Reply:
That refactor looks good, thanks!